test/service/FileTransferService.test.ts

```diff
@@ -0,0 +1,69 @@
+import { expect } from 'chai';
+import FormData from 'form-data';
+import nock = require('nock');
+
+import { FileTransferService } from 'app/service/FileTransferService'
+
+describe('FileTransferService', () => {
+
+    const KEY: string = 'mockKey';
+    const HOST: string = 'http://localhost';
+    const evidenceUploadService = new FileTransferService(HOST , KEY);
+
+    describe('upload file', () => {
+        const URI: string = '/dev/files';
+
+        it('should throw an error when evidence not defined', () => {
+
+            [undefined, null].forEach(async evidence => {
+                try {
+                    await evidenceUploadService.upload(evidence!, 'filename')
+                } catch (err) {
+                    expect(err).to.be.instanceOf(Error)
+                        .and.to.haveOwnProperty('message').equal('Evidence file is missing')
+                }
+            })
+        });
+
+        it('should throw an error when file name not defined', () => {
+
+            const evidence: Buffer = Buffer.from('This is a test');
+
+            [undefined, null].forEach(async filename => {
+                try {
+                    await evidenceUploadService.upload(evidence as Buffer, filename!)
+                } catch (err) {
+                    expect(err).to.be.instanceOf(Error)
+                        .and.to.haveOwnProperty('message').equal('File name is missing')
+                }
+            })
+        });
+
+
+        it('should return 201 status code when evidence uploaded', async () => {
```

Comment:
Upload method should handle 415 response code gracefully which is how API communicates that type of the uploaded file is not supported.

Please add test for at least 415 response code as per https://github.com/companieshouse/file-transfer-api/blob/master/specs/schema.json#L56

Reply:
PS. I am not sure what 404 scenario is, seems bit uncommon to have `the file does not exist` scenario on resource creation.